### PR TITLE
Fixed some queries

### PR DESCRIPTION
**Pull request by SilasMarvin**

### CODE REVIEW

File: pgml-sdks/pgml/src/collection.rs [hunk 1]
```diff
@@ -936,6 +937,8 @@ impl Collection {
                 .fetch_all(&pool)
                 .await;
 
+                println!("\n\n{result:?}\n\n");
```
Comment (levkk) [inline pgml-sdks/pgml/src/collection.rs:940]: Remove this, it'll spam our users logs.

File: pgml-sdks/pgml/src/lib.rs [hunk 2]
```diff
@@ -354,18 +354,21 @@ mod tests {
...
         collection
-            .upsert_documents(generate_dummy_documents(3))
+            .upsert_documents(generate_dummy_documents(5))
             .await?;
         let results = collection
             .vector_search("Here is some query", &mut pipeline, None, None)
             .await?;
-        assert!(results.len() == 3);
+
+        println!("\n{results:?}\n");
```
Comment (levkk) [inline pgml-sdks/pgml/src/lib.rs:369]: Same.

File: pgml-sdks/pgml/src/collection.rs (hunk 1 above)
Comment (SilasMarvin) [reply pgml-sdks/pgml/src/collection.rs:940]: Oh wow great catches sorry

File: pgml-sdks/pgml/src/queries.rs [hunk 3]
```diff
@@ -207,31 +214,20 @@ FROM
...
-  INNER JOIN %s chunks ON chunks.id = comparison.chunk_id 
-  INNER JOIN %s documents ON documents.id = chunks.document_id 
-  ORDER BY 
-  comparison.score DESC 
-  LIMIT 
+  %s embeddings
+INNER JOIN %s chunks ON chunks.id = embeddings.chunk_id 
+INNER JOIN %s documents ON documents.id = chunks.document_id 
+ORDER BY 
+  embeddings.embedding <=> $1::vector ASC
```
Comment (levkk) [inline pgml-sdks/pgml/src/queries.rs:229]: This uses the index but does it mean the same? The sort on the right sorts by highest scores first, while this scores by lowest scores first. In fact, won't this give the worst results by similarity?

Comment (SilasMarvin) [reply pgml-sdks/pgml/src/queries.rs:229]: Yes it means the same. <=> is the cosine distance function. We want ones with the lowest distance. 1 - <=> is the cosine similarity. You want ones with high similarity which is the same as low distance

File: pgml-sdks/pgml/src/queries.rs [hunk 4]
```diff
@@ -134,8 +134,10 @@ WHERE
   AND id NOT IN (
```
Comment (montanalow) [inline pgml-sdks/pgml/src/queries.rs:134]: Exists is more idiomatic
```suggestion
  AND NOT EXISTS (
```

File: pgml-sdks/pgml/src/queries.rs [hunk 5]
```diff
@@ -157,8 +159,10 @@ WHERE
...
```
Comment (montanalow) [inline pgml-sdks/pgml/src/queries.rs:159]:
```suggestion
  AND NOT EXISTS (
```

File: pgml-sdks/pgml/src/queries.rs (hunk 4 above)
Comment (montanalow) [reply pgml-sdks/pgml/src/queries.rs:134]: The query planner should optimize this as equivalent, but you could also do a `LEFT OUTER JOIN "table" ON chunk_id = ... WHERE "table".id IS NULL` which will select only the missing rows from "table", if the subquery shows up as an expensive component.

**Pull request closed without merging**